sycl/source/detail/pi.cpp

```diff
@@ -386,8 +690,25 @@ static void initializePlugins(std::vector<plugin> *Plugins) {
       std::cerr << "SYCL_PI_TRACE[basic]: "
                 << "Plugin found and successfully loaded: "
                 << PluginNames[I].first << std::endl;
-  }
 
+    fillPlatformAndDeviceCache(Plugins->back());
+  } // end of for
+
+  // The host platform should always be available unless not allowed by the
+  // SYCL_DEVICE_FILTER
+  detail::device_filter_list *FilterList =
+      detail::SYCLConfig<detail::SYCL_DEVICE_FILTER>::get();
+  if (!FilterList || FilterList->containsHost()) {
+    std::vector<PlatformImplPtr> &PlatformCache =
+        GlobalHandler::instance().getPlatformCache();
+    std::vector<DeviceImplPtr> &DeviceCache =
+        GlobalHandler::instance().getDeviceCache();
+    PlatformImplPtr PlatformImpl = platform_impl::getHostPlatformImpl();
+    PlatformCache.emplace_back(PlatformImpl);
+    std::shared_ptr<device_impl> Device =
+        platform_impl::getOrMakeHostDeviceImpl();
+    DeviceCache.emplace_back(Device);
```

Comment:
Shouldn't this be done with mutexes for `PlatformCache` and `DeviceCache` locked.

Reply:
Yes. 
Done. Thanks.

Reply:
Could you please avoid situation when both mutexes are locked? It seems adding a device to the device cache doesn't require the platform mutex to be locked and vice versa.